### PR TITLE
Concrete Mempools

### PR DESCRIPTION
**Pull request by fosskers**

### CODE REVIEW

File: test/Chainweb/Test/Mempool/RestAPI.hs [hunk 1]
```diff
@@ -106,12 +106,8 @@ newPool = Pool.createPool newTestServer destroyTestServer 1 10 20
...
 ------------------------------------------------------------------------------
 
-serverMempools
-    :: [(ChainId, MempoolBackend t)]
-    -> ChainwebServerDbs t a RocksDbCas {- ununsed -}
-serverMempools mempools = emptyChainwebServerDbs
-    { _chainwebServerMempools = mempools
-    }
+serverMempools :: ChainwebServerDbs a RocksDbCas {- ununsed -}
+serverMempools = emptyChainwebServerDbs
```
Comment (fosskers) [inline test/Chainweb/Test/Mempool/RestAPI.hs:110]: Hmm...

File: src/Chainweb/RestAPI.hs [hunk 2]
```diff
@@ -248,16 +244,14 @@ someChainwebServer v dbs mr (HeaderStream hs) (Rosetta r) =
     pacts = _chainwebServerPactDbs dbs
     cuts = _chainwebServerCutDb dbs
     peers = _chainwebServerPeerDbs dbs
-    mempools = _chainwebServerMempools dbs
     concreteMs = map (second (_chainResMempool . snd)) pacts
```
Comment (larskuhtz) [inline src/Chainweb/RestAPI.hs:247]:
```suggestion
    mempools = map (second (_chainResMempool . snd)) pacts
```

File: src/Chainweb/RestAPI.hs [hunk 3]
```diff
@@ -236,7 +232,7 @@ someChainwebServer v dbs mr (HeaderStream hs) (Rosetta r) =
         <> maybe mempty (someSpvServers v) cuts
         <> somePayloadServers v payloads
         <> someBlockHeaderDbServers v blocks
-        <> Mempool.someMempoolServers v mempools
+        <> Mempool.someMempoolServers v concreteMs
```
Comment (larskuhtz) [inline src/Chainweb/RestAPI.hs:235]:
```suggestion
        <> Mempool.someMempoolServers v mempools
```

File: test/Chainweb/Test/Mempool/RestAPI.hs [hunk 4]
```diff
@@ -70,7 +70,7 @@ newTestServer = mask_ $ do
     server inMemCfg inmemMv envMv restore =
         InMem.withInMemoryMempool inMemCfg version $ \inmem -> do
             putMVar inmemMv inmem
-            restore $ withTestAppServer True version (return $! mkApp inmem) mkEnv $ \env -> do
+            restore $ withTestAppServer True version (return mkApp) mkEnv $ \env -> do
```
Comment (fosskers) [inline test/Chainweb/Test/Mempool/RestAPI.hs:73]: This is likely the cause of the failures.

Comment (fosskers) [reply test/Chainweb/Test/Mempool/RestAPI.hs:73]: ... but why? The actual chainweb lib no longer uses those "generic" mempools anymore...

File: test/Chainweb/Test/Mempool/RestAPI.hs (hunk 1 above)
Comment (larskuhtz) [reply test/Chainweb/Test/Mempool/RestAPI.hs:110]: yeah, that's probably not correct ;-)

File: test/Chainweb/Test/Mempool/RestAPI.hs (hunk 4 above)
Comment (larskuhtz) [inline test/Chainweb/Test/Mempool/RestAPI.hs:73]: Reverting this should fix the issue.

**Comment by fosskers** [comment]
Abandoning the approach.

**Pull request closed without merging**